enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,493 @@
+Furthermore, it will reduce the operational complexity of running a managed service since the Managed Tenant SREs will not need to familiarise themselves with a new monitoring stack for each managed service.
+In future it can be used to allow customers to define their monitoring stacks. UWM could use it under the hood too. In details:
+
+* Provide a way for managed services to scrape metrics from their own services and from platform components into prometheus-compatible storage.
+* Allow managed service owners to query metrics from their services and platform components from prometheus-compatible storage.
+* Provide a way for managed services to define alerting and recording rules on their own metrics and a subset of platform metrics.
+* Provide a way for managed services to create dashboards for monitoring their services.
+* Allow aggregating portions of managed service metrics at a multi-cluster level, e.g. for reporting SLOs.
+* Allow longer retention for portions of data (e.g. calculating 1 month burn down SLO budget).
+* Allow analytics on portion of managed service data in conjuction with Telemeter data at the aggregation level.
+* End customers using UWM should not see managed service metrics.
+* Soft Goal: A foundation for next iterations mentioned in ### Long term motivation.
+
+### Long Term Motivation
+
+To avoid maintaining multiple monitoring solutions, this proposal aims to be a foundation for future monitoring enhancements. Therefore, the following goals are left out of scope for the current enhancement but are kept in mind for future iterations:
+
+* Such a solution as a community project which is easy to use & contribute to.
+* Reduce Platform Monitoring overhead and its single point of failure (SPOF) property.
+* Auto-scaling metric scraping based on number of ServiceMonitors
+* Soft multi-tenancy for local storage (TSDB).
+* Opt-in for full forwarding mode.
+* Full architecture for Hypershift ([WIP](https://docs.google.com/document/d/1BJarERppgiJ8esc6d8anbJMOQ0AflFBQea-Zc9wAp0s/edit))
+
+### Non-Goals
+
+* Access controls for managed service owners for metrics from different managed services. For example, there are currently no requirements for owners of managed services A to be prohibited from seeing metrics from managed service B, and vice versa.
+* Expose managed service metrics to end-users. We know e.g. OCS addon wants this, but this can be achieved on top of the proposed solution using client or backend side filtering for data isolation if needed (e.g. prom-label-proxy). The monitoring team can do this in the next iteration.
+* Integrate managed service metrics into the OpenShift console. We know OCS wants to integrate with OCP Console, but we assume the console team can help support switching between Prometheus-compatible data sources.
```

Comment:
> Do you mean Thanos Querier? If yes, then we assume the PM one? (Platform Monitoring)
Yes I meant "Querier" not "ruler"

> We believe there should be an effort for OpenShift Console to discover MonitorStacks that comes from addons that need extra widget/dashboard functionality on console based on that and use that endpoint instead of PM Thanos Querier.
My understanding is that this is the exact thing they object to as most of the time you will need some of the UI to work against one querier while some other part of the UI against another. And the console was not built to do that. 
For UWM they just use the same querier as they use for PM

My position is that if we will wait for them to do this kind of work it might ship far far into the future. I think this proposal needs to have a goal that says that this new monitoring stack needs to expose whatever is needed for remote writes to the Thanos instance used by the console. I think that most of the effort will go into autentication